File: src/signal_simple/signal_simple.erl
-module(signal_simple).

% elp:ignore W0012 W0040 E1599
-moduledoc {file, "../../docs/signal_simple.md"}.

-export([
    start_then_signal/0,
    start_with_signal/0
]).

-define(CLUSTER, cluster_1).
-define(CANCEL_ACTIVITY_SIGNAL, "cancel_activity").

start_then_signal() ->
    {ok, #{workflow_execution := WE}} =
        temporal_sdk:start_workflow(?CLUSTER, "default", signal_simple_workflow),
    timer:sleep(1_000),
    temporal_sdk:signal_workflow(?CLUSTER, WE, ?CANCEL_ACTIVITY_SIGNAL),
    temporal_sdk:wait_workflow(?CLUSTER, WE).

start_with_signal() ->
    temporal_sdk:start_workflow(?CLUSTER, "default", signal_simple_workflow, [
        wait, {signal_name, ?CANCEL_ACTIVITY_SIGNAL}
    ]).

File: lib/signal_simple/signal_simple.ex
defmodule SignalSimple do
  @external_resource "docs/signal_simple.md"
  @moduledoc TemporalSdk.Utils.exdoc!("docs/signal_simple.md")

  @cluster :cluster_1
  @cancel_activity_signal "cancel_activity"

  def start_then_signal() do
    {:ok, %{workflow_execution: we}} =
      TemporalSdk.start_workflow(@cluster, "default", SignalSimple.Workflow)

    Process.sleep(1_000)
    TemporalSdk.signal_workflow(@cluster, we, @cancel_activity_signal)
    TemporalSdk.wait_workflow(@cluster, we)
  end

  def start_with_signal() do
    TemporalSdk.start_workflow(@cluster, "default", SignalSimple.Workflow, [
      :wait,
      signal_name: @cancel_activity_signal
    ])
  end
end
